From 954b523a998933b1edbe9af3fe7b1c74595cf148 Mon Sep 17 00:00:00 2001
From: Lorenzo Dal Col <lorenzo@khronosgroup.org>
Date: Tue, 17 Sep 2024 17:18:20 +0200
Subject: [PATCH] Allow running the CTS with unknown versions of Vulkan in the
 driver

This will default to VK_API_MAX_FRAMEWORK_VERSION, which at this time is 1.3.x.
VK-GL-CTS issue: 5165
Components: Framework, Vulkan
Affects: None

Change-Id: I954d0144db9ff5f3848c81df61dd263505b17ac2
Upstream-Status: Backport [https://github.com/KhronosGroup/VK-GL-CTS/commit/609c09b3163d86d7e4819fbfc00a616e364f14f0]
Signed-off-by: Randolph Sapp <rs@ti.com>
---
 external/vulkancts/framework/vulkan/vkApiVersion.cpp | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/external/vulkancts/framework/vulkan/vkApiVersion.cpp b/external/vulkancts/framework/vulkan/vkApiVersion.cpp
index d996cb0e6..8720df52a 100644
--- a/external/vulkancts/framework/vulkan/vkApiVersion.cpp
+++ b/external/vulkancts/framework/vulkan/vkApiVersion.cpp
@@ -134,7 +134,14 @@ deUint32 minVulkanAPIVersion(deUint32 lhs, deUint32 rhs)
 	for (auto it = begin(commonPredecessors); it != end(commonPredecessors); ++it)
 		if (isApiVersionPredecessor(rhs, *it) && isApiVersionPredecessor(lhs, *it))
 			return *it;
-	return 0;
+
+#ifndef CTS_USES_VULKANSC
+    // If we got to this point, it means we are dealing with an unknown version.
+    // We assume it to be valid, and we default to VK_API_MAX_FRAMEWORK_VERSION which is generated from the spec.
+    return VK_API_MAX_FRAMEWORK_VERSION;
+#else
+    return VKSC_API_MAX_FRAMEWORK_VERSION;
+#endif
 }
 
 } // vk
-- 
2.50.1

